src/utilities/modules/vaccines.js

```diff
@@ -158,10 +187,16 @@ export const sensorLogsExtractBatches = ({ sensorLogs = [], itemBatch, item } =
       batches: Object.values(batches).map(batchObject => batchObject),
     };
   });
+  // Get all ItemBatch objects associated with these sensorLogs, used in
+  // calculating statistics of this breach.
+  const allItemBatches = database
+    .objects('ItemBatch')
+    .filtered(sensorLogs.map(({ id }) => `sensorLogs.id = "${id}"`).join(' OR '));
```

Comment:
you've mentioned you forgot to account for surrounding batches.
Just a filter at the start of this method should do it (filter by isInBreach). 

Method should still return unfiltered sensor logs though.

Reply:
89e3f18862158caacbebe4e4f2ce61fdba90047c